modules/new_profile/jsx/NewProfileForm.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * New Profile Form
+ *
+ * Create a new proflie form
+ *
+ * @author  Shen Wang
+ * @version 1.0.0
+ * */
+class NewProfileForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            data: {
+            },
+            configData: {},
+            formData: {},
+            newData: {},
+            isLoaded: false,
+            isCreated: false,
+            errMessage: '',
+        };
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.setFormData = this.setFormData.bind(this);
+        this.fetchData = this.fetchData.bind(this);
+    }
+
+    /**
+   * Retrive data from the provided URL and save it in state
+   * Additionaly add hiddenHeaders to global loris vairable
+   * for easy access by columnFormatter.
+   */
+    fetchData() {
+        $.ajax(
+            loris.BaseURL + '/new_profile/?format=json',
+            {
+                method: 'GET',
+                dataType: 'json',
+                success: (data) => {
+                    // FIXME: Remove the following line of code as soon as hiddenHeaders is
```

Comment:
This comment can be removed since it's only relevant to hidden headers

Reply:
done